hera_qm/ant_metrics.py

```diff
@@ -0,0 +1,394 @@
+        return crossPolRatio
+    else:
+        return per_antenna_modified_z_scores(crossPolRatio)
+
+def average_abs_metrics(metrics1, metrics2):
+    '''Averages the absolute value of two metrics together.'''
+    
+    if set(metrics1.keys()) != set(metrics2.keys()):
+        raise KeyError('Metrics being averaged have differnt (ant,antpol) keys.')
+    return {key: np.abs(metrics1[key]/2) + np.abs(metrics2[key]/2) for key in metrics1.keys()}
+
+def load_antenna_metrics(metricsJSONFile):
+    '''Loads all cut decisions and meta-metrics from a JSON into python dictionary.'''
+    
+    with open(metricsJSONFile,'r') as infile:
+        jsonMetrics = json.load(infile)
+    return {key: eval(str(val)) for key,val in jsonMetrics.items()}
+
+
+#######################################################################
+#High level functionality for HERA
+#######################################################################
+
+
+class Antenna_Metrics():
+    '''Object for holding relevant visibility data and metadata with interfaces to four 
+    antenna metrics (two for identifying dead antennas, two for identifying cross-polarized ones), 
+    an iterative method for identifying one bad antenna at a time while keeping track of all 
+    metrics, and for writing metrics to a JSON.'''
+    
+    def __init__(self, dataFileList, reds, fileformat='miriad'):
+        '''Arguments:
+        dataFileList -- List of data filenames for the different polarizations
+        reds -- List of lists of tuples of antenna numbers that make up redundant baseline groups
+        format -- default miriad. Otheqr options: uvfits
+        '''
+        
+        self.data = UVData()
+        if fileformat is 'miriad':
+            self.data.read_miriad(dataFileList)
+        elif fileformat is 'uvfits':
+            self.data.read_uvfits(dataFileList)
+        else:
+            raise ValueError('Unrecognized file format' + str(fileformat))
+        self.ants = self.data.get_ants()
+        self.pols = [pol.lower() for pol in self.data.get_pols()]
+        self.antpols = [antpol.lower() for antpol in self.data.get_feedpols()]
+        self.bls = list(set([(i,j) for (i,j,pol) in self.data.get_antpairpols()]))
+        self.reds = reds
+
+        # For using data containers until pyuvdata gets faster
+        # from hera_cal import firstcal
+        # from hera_qm.datacontainer import DataContainer
+        # self.uvdata = self.data
+        # datapack, flagspack = firstcal.UVData_to_dict([self.uvdata])
+        # self.data = DataContainer(datapack)
+
+        if len(self.antpols) is not 2 or len(self.pols) is not 4:
+            raise ValueError('Missing polarization information. pols =' + str(self.pols) + ' and antpols = ' + str(self.antpols))
+
+    def mean_Vij_metrics(self, pols=None, xants=[], rawMetric=False):
+        '''Local wrapper for mean_Vij_metrics in hera_qm.ant_metrics module.'''
+
+        if pols is None:
+            pols = self.pols
```

Comment:
I could go either way on this. `xants` is something that changes during the lifetime of an `Antenna_Metrics` object while pols generally doesn't. I think I'll leave it as is.